src/types/storeTypes.ts

```diff
@@ -1,21 +1,21 @@
-import { Feature } from "geojson";
-import { AppConfig, BoundingBox, FeatureIndicators, Geojson } from "./resourceTypes";
+import { AppConfig, BoundingBox, FeatureIndicators, MapFeature, MapSettings } from "./resourceTypes";
 import { Dict } from "./utilTypes";
 
 export interface AppState {
-    selectedIndicator: string;
-
-    // When a country is selected (when value is non-empty), we display admin2 data for it.
-    selectedCountryId: string;
-
+    // these state variables only get updated once on initialisation, we keep all data in
+    // dictionaries with country ids as keys, mirroring the resources on disk
     appConfig: AppConfig | null;
-
     countryBoundingBoxes: Dict<BoundingBox>;
-
-    // We keep all data in dictionaries with country ids as keys, mirroring the resources on disk
     admin1Indicators: Dict<FeatureIndicators>;
-    admin1Geojson: Dict<Geojson>;
+    admin1Geojson: Dict<MapFeature[]>;
+    initialisationComplete: boolean;
+
+    // these are initialised as empty objects and extended every time the user selects a
+    // country
     admin2Indicators: Dict<FeatureIndicators>;
-    admin2Geojson: Dict<Geojson>;
-    admin0GeojsonFeature: Feature | null;
+    admin2Geojson: Dict<MapFeature[]>;
+
+    // these variables change every time there is an update to the map
+    mapSettings: MapSettings | null;
+    admin0GeojsonFeature: MapFeature | null;
```

Comment:
Have the feature dictionaries actually changed, or were these just wrongly defined before?

Reply:
i havent changed any structure of these, they were just wrongly defined i think, my IDE makes the file title red and red squiggles everywhere if it is the wrong type, does your IDE not do that?

Reply:
It does but it sometimes gets it wrong, so I don't always pay much attention to it! 😆